lib/cpp/base_model/model_generalized_linear.cpp

```diff
@@ -2,18 +2,32 @@
 
 #include "tick/base_model/model_generalized_linear.h"
 
-ModelGeneralizedLinear::ModelGeneralizedLinear(const SBaseArrayDouble2dPtr features,
-                                               const SArrayDoublePtr labels,
-                                               const bool fit_intercept,
-                                               const int n_threads)
-  : ModelLabelsFeatures(features, labels),
-    n_threads(n_threads >= 1 ? n_threads : std::thread::hardware_concurrency()),
+template <class T, class K>
```

Comment:
Do we need 2 templates? Or is this only for atomics? 
Also; K seems to be the "main" template and T only used if we have atomic. I find this confusing and would rather have T as "main" template and K set by default to T (if that's possible).